src/coreclr/vm/comdelegate.cpp

```diff
@@ -330,6 +339,73 @@ BOOL AddNextShuffleEntryToArray(ArgLocDesc sArgSrc, ArgLocDesc sArgDst, SArray<S
     {
         // We should have slots to shuffle in the destination at the same time as the source.
         _ASSERTE(iteratorDst.HasNextOfs());
+#if defined(TARGET_LOONGARCH64)
```

Comment:
Can this logic be in `ShuffleIterator` iterator instead? All other platforms have the platform-specific logic in `ShuffleIterator`.

Reply:
> Can this logic be in `ShuffleIterator` iterator instead? All other platforms have the platform-specific logic in `ShuffleIterator`.

Yes, this is needed for LoongArch64's ABI.
Liking LoongArch64's ABI, we have to get the details info of the struct's fields, where the other architectures now, liking X64/arm64, are not need these.

Reply:
Can the special casing of `LoongArch64` be done inside the ShuffleIterator ?

For example, Linux x64 ABI (`UNIX_AMD64_ABI`) is even more complex than LoongArch64 and these details are hidden inside the `ShuffleIterator`.

Reply:
I will think about it.

Reply:
@jkotas   Sorry, I forgot this comment.
Thanks, I will update it later.

Reply:
I will delete this as the logic condition of the two `if statement` are confict which is not used for LoongArch64 now.
```
        if (!sArgDst.m_byteStackSize && (sArgSrc.m_structFloatFlag > 0))
        {
            if (sArgSrc.m_byteStackSize > 0)
```